xds/src/main/java/io/grpc/xds/CdsLoadBalancer2.java

```diff
@@ -238,14 +312,19 @@ private ClusterState(String name) {
       }
 
       private void start() {
+        shutdown = false;
         xdsClient.watchXdsResource(XdsClusterResource.getInstance(), name, this);
       }
 
       void shutdown() {
         shutdown = true;
         xdsClient.cancelXdsResourceWatch(XdsClusterResource.getInstance(), name, this);
-        if (childClusterStates != null) {  // recursively shut down all descendants
-          for (ClusterState state : childClusterStates.values()) {
+        if (childClusterStates == null) {
+          return;
+        }
```

Comment:
Why add this short circuit? It's better to have a single return point from a method and this change doesn't seem to affect the logic.

Reply:
Was trying to reduce nesting `if { for { if { } } }`. Rewrote it to use a stream instead of a for with an if.